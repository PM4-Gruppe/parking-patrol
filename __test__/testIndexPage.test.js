import Home from '../src/pages/index'
import renderer from 'react-test-renderer'
import React from 'react'

jest.mock('next/link', () => {
    const Link = ({ children, ...props }) => (
        <a {...props}>{children}</a>
    );
    Link.displayName = 'Link';
    return Link;
});

describe('Home', () => {
    it('should match snapshot', () => {
        const tree = renderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});