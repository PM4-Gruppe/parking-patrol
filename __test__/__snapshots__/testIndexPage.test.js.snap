// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match snapshot 1`] = `
<div
  className="mb-auto"
>
  <div
    className="flex flex-col center justify-evenly m-10"
  >
    <div
      className="p-5"
    >
      <div
        className="bg-neutral-500/50 rounded-2xl"
      >
        <div
          className="bg-green-700 p-3 text-xl text-white rounded-t-2xl"
        >
          Aktuelle Parkzone
        </div>
        <div
          className="p-3"
        >
          <div>
            <p
              className="text-white text-lg"
            >
              aktuell in keiner Parkzone
            </p>
            <p
              className="text-neutral-400 text-sm m-0 p-0"
            >
              n√§chste Parkzone Eulachpassage ist 300m entfernt
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      className="flex flex-row center justify-between"
    >
      <div
        className="w-1/2 p-5"
      >
        <div
          className="bg-neutral-500/50 rounded-2xl"
        >
          <div
            className="bg-green-700 p-3 text-xl text-white rounded-t-2xl"
          >
            Erfasste Fahrzeuge
          </div>
          <div
            className="p-3"
          >
            <p
              className="text-white text-lg"
            >
              192 Fahrzeuge
            </p>
          </div>
        </div>
      </div>
      <div
        className="w-1/2 p-5"
      >
        <div
          className="bg-neutral-500/50 rounded-2xl"
        >
          <div
            className="bg-green-700 p-3 text-xl text-white rounded-t-2xl"
          >
            Offene Vergehen
          </div>
          <div
            className="p-3"
          >
            <p
              className="text-white text-lg"
            >
              23 Vergehen
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="flex justify-center w-100 h-full"
    id="main"
  >
    <div
      className="absolute inset-x-0 bottom-0 h-28 bg-neutral-900"
    />
    <a
      className="absolute bottom-16"
      href="/camera"
    >
      <div
        className="bg-green-700 hover:bg-green-800 rounded-full p-5 border-solid border-8 border-neutral-800"
      >
        <svg
          className="w-20 h-20 p-1 text-white"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 8c-2.168 0-4 1.832-4 4s1.832 4 4 4 4-1.832 4-4-1.832-4-4-4zm0 6c-1.065 0-2-.935-2-2s.935-2 2-2 2 .935 2 2-.935 2-2 2z"
          />
          <path
            d="M20 5h-2.586l-2.707-2.707A.996.996 0 0 0 14 2h-4a.996.996 0 0 0-.707.293L6.586 5H4c-1.103 0-2 .897-2 2v11c0 1.103.897 2 2 2h16c1.103 0 2-.897 2-2V7c0-1.103-.897-2-2-2zM4 18V7h3c.266 0 .52-.105.707-.293L10.414 4h3.172l2.707 2.707A.996.996 0 0 0 17 7h3l.002 11H4z"
          />
        </svg>
      </div>
    </a>
    <a
      className="absolute bottom-5 left-20 rounded-full items-center bg-neutral-900 hover:bg-neutral-700"
      href="https://en.wikipedia.org/wiki/Settings_(Windows)"
    >
      <svg
        className="w-20 h-20 p-4 text-white"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12"
          cy="12"
          r="3"
        />
        <path
          d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
        />
      </svg>
    </a>
    <a
      className="absolute bottom-6 right-20 rounded-full items-center bg-neutral-900 hover:bg-neutral-700"
      href="/carlist"
    >
      <svg
        className="w-20 h-20 p-4 text-white"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M48 48a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm0 160a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm0 160a48 48 0 1 0 48 48 48 48 0 0 0-48-48zm448 16H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0-320H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16V80a16 16 0 0 0-16-16zm0 160H176a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h320a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
        />
      </svg>
    </a>
  </div>
</div>
`;
